### PR TITLE
feat(mui-datepicker): add datepicker component

### PR DESCRIPTION
**Pull request by jordan-a-young**
Creates the `Datepicker` component with styling from UX

### CODE REVIEW

**Review by gregmartDOTin — APPROVED** [review]

File: packages/theme/src/lib/light-theme.ts [hunk 1]
```diff
@@ -598,6 +627,7 @@ export const lightTheme = {
       styleOverrides: {
         root: {
           '&.MuiInputBase-root': {
+            padding: '8px 12px',
```
Comment (LauRoxx) [inline packages/theme/src/lib/light-theme.ts:630]: Have we identified all the effects of moving the padding from the input to the root? Looks like we're still using the `adornedEnd` etc items that alter the root vs swapping to `inputAdornedEnd` etc to ensure it all works correctly together.

I'm a bit nervous to move away from MUI patterns with a base class like this that could be difficult to change and account for all effects later down the line.

File: packages/theme/src/lib/light-theme.ts [hunk 2]
```diff
@@ -521,14 +546,17 @@ export const lightTheme = {
...
       styleOverrides: {
-        root: {
-          marginLeft: '0px',
-          '&:not(&.Mui-error) .MuiSvgIcon-root': {
-            display: 'none',
-          },
-          '.MuiSvgIcon-root': {
-            verticalAlign: 'text-bottom',
-          },
+        root: ({ ownerState }: { ownerState: { disabled: boolean } }) => {
```
[the remark this answers is not in the document]
Comment (jordan-a-young) [reply packages/theme/src/lib/light-theme.ts:549]: It wasn't working without it. I can't remember if I tested it again after bringing in your TextField changes. I can double check

File: packages/theme/src/lib/light-theme.ts (hunk 1 above)
Comment (jordan-a-young) [reply packages/theme/src/lib/light-theme.ts:630]: I had to do this to get it to work with the Password Field example if I remember correctly. I hadn't checked it again after bringing in your changes. I'll see if it works without it

File: packages/theme/src/lib/light-theme.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jordan-a-young) [reply packages/theme/src/lib/light-theme.ts]: This wasn't needed anymore so I removed the change.

File: packages/theme/src/lib/light-theme.ts [hunk 3]
```diff
@@ -539,6 +567,7 @@ export const lightTheme = {
           display: important('flex'),
           flexDirection: important('row-reverse'),
           justifyContent: important('flex-end'),
+          marginBottom: '3px',
```
[the remark this answers is not in the document]
Comment (jordan-a-young) [reply packages/theme/src/lib/light-theme.ts:567]: Didn't look like it changed alignment. I added the `helpTopicId` to available props and to the `Datepicker` story so you can double check.

File: packages/theme/src/lib/light-theme.ts (hunk 1 above)
Comment (jordan-a-young) [reply packages/theme/src/lib/light-theme.ts]: I moved where the padding is applied. Changed it to use `MuiInputBase-input` and updated the padding for `MuiInputBase-adornedEnd` to match our padding of `12px`.

File: packages/theme/src/lib/light-theme.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (LauRoxx) [reply packages/theme/src/lib/light-theme.ts:567]: it messes with it a little, but I can also put in the fix during my autocomplete ticket this sprint if we want to finish this up.

**Pull request merged**